File: tests/syntax/supported/perl/highlight.pl
#!/usr/bin/env perl
# Highlighting sampler
use strict;
use warnings;
package Sample;

my $COUNT = 42;
my $pattern = qr/^item-(\d+)$/i;

sub greet {
	my ($name) = @_;
	say "Hello, $name";
	return $COUNT if $name =~ $pattern;
}

my @items = map { $_ * 2 } (1, 2, 3);
my %lookup = (answer => 42);
greet("world");

File: tests/syntax/supported/swift/highlight.swift
/// Highlighting sampler
import Foundation

struct Point {
    let x: Int
    var y: Int

    func distance(to other: Point) -> Double {
        let dx = Double(x - other.x)
        return dx
    }
}

public func greet(name: String) -> String {
    let message = "Hello, \(name)"
    return message
}

let ready = true
